### ISSUE TITLE
Further details about optimization process

### ISSUE THREAD

**Issue by Frankey419** [entitize/Fakeddit#8]
Hi Meywether,

- We extracted features from (VGG,BERT,etc.) for specific size vectors. The whole dataset was processed. The images were resized to fit the input size vector for the image neural network. For example, VGG16 has an input size of 224x224, so we scaled the images to 224x224. For the text, we applied filtering as described in the paper such as removing punctuation, revealing words, etc. These features were then used for classification. 

- We utilized separate image and text models for feature extraction (VGG,BERT,etc.). Once all features have been extracted we trained a separate model to combine these extracted features and classify our input.

Does this answer your questions?

_Originally posted by @entitize in https://github.com/entitize/Fakeddit/issues/5#issuecomment-619213348_

**Comment by Frankey419** [comment]
Dear entitize,
May I ask for further details about the optimization process:
1. As you mentioned above, I thought the Bert model or ResNet model is not fine-tuned in the optimization process. So, did you have fine-tuned them before the feature extraction, or just utilize the open-source pre-trained models for the feature extraction?
2. There are three different classification tasks, did you complete them with a single model in a multitasking manner, or you trained a model for each task?

By the way, could you provide more information about the structure of ResNet+Bert_large model? such as the number of units in the hidden layer, and the learning rate. Sharing the codes are better.
Best regards

**Comment by entitize** [comment]
Hi Frankey, 

We utilized the open source pertained models for feature extraction. For example, as described in the paper, for BERT features we used bert-as-service to extract features. 

2. We trained a model for each task: trained on image only, trained on text only, and trained on image + text. 

The structure of the model including the number of units in the hidden layer n and the learning rate is described in detail in the Results section of the paper.

**Comment by Frankey419** [comment]
> Hi Frankey,
> 
> We utilized the open source pertained models for feature extraction. For example, as described in the paper, for BERT features we used bert-as-service to extract features.
> 
> 1. We trained a model for each task: trained on image only, trained on text only, and trained on image + text.
> 
> The structure of the model including the number of units in the hidden layer n and the learning rate is described in detail in the Results section of the paper.

In fact, for the 2nd question, I mean the 2-way-classification, 3-way-classification, and 6-way-classification.
Besides, I tried to utilize the bert_embeddings and ResNet features, but it seems too weak to achieve the reported results(0.8929 for 2-way-classification on the validation dataset). Could you provide the baseline source code?

**Comment by entitize** [comment]
[classify_final2.py.zip](https://github.com/entitize/Fakeddit/files/5279768/classify_final2.py.zip)

**Comment by entitize** [comment]
Also, note that the results in the paper are based on samples that have both text and image (only multimodal samples).
